### PR TITLE
Fix colour issue

### LINKED ISSUE #2388: Path color is not the same as the others
I've noticed a strange bug while experimenting with paths where the path color does not match any other drawable's color

Here is a test script to recreate this problem:
```cs
Children = new Drawable[]
{
    new TexturedPath
    {
        Anchor = Anchor.TopRight,
        Origin = Anchor.TopRight,
        Vertices = new List<Vector2> { Vector2.One* 50, Vector2.One* 500 },
        Colour = new Color4(238, 93, 146, 255),
    },
    new Box
    {
        Size = new Vector2(500),
        Colour = new Color4(238, 93, 146, 255),
    }
};
```

### PR DESCRIPTION
**Pull request by Altenhh**
resolves #2388

### CODE REVIEW

File: osu.Framework/Graphics/Lines/Path_DrawNode.cs [hunk 1]
```diff
@@ -56,8 +56,8 @@ public override void ApplyState()
             private Vector2 relativePosition(Vector2 localPos) => Vector2.Divide(localPos, drawSize);
 
             private Color4 colourAt(Vector2 localPos) => DrawColourInfo.Colour.HasSingleColour
-                ? (Color4)DrawColourInfo.Colour
-                : DrawColourInfo.Colour.Interpolate(relativePosition(localPos)).Linear;
+                ? DrawColourInfo.Colour.Interpolate(relativePosition(localPos)).Linear
```
Comment (Tom94) [inline osu.Framework/Graphics/Lines/Path_DrawNode.cs]: This change does not make much sense. Why would is make sense to interpolate when the colour only has a single entry?

I think the problem lies fundamentally elsewhere.

File: osu.Framework/Graphics/Lines/Path_DrawNode.cs [hunk 2]
```diff
@@ -56,8 +56,8 @@ public override void ApplyState()
             private Vector2 relativePosition(Vector2 localPos) => Vector2.Divide(localPos, drawSize);
 
             private Color4 colourAt(Vector2 localPos) => DrawColourInfo.Colour.HasSingleColour
-                ? (Color4)DrawColourInfo.Colour
```
Comment (Tom94) [inline osu.Framework/Graphics/Lines/Path_DrawNode.cs]: I suspect the correct fix for the problem is to append `.Linear` here.

File: osu.Framework/Graphics/Lines/Path_DrawNode.cs [hunk 3]
```diff
@@ -55,9 +56,7 @@ public override void ApplyState()
 
             private Vector2 relativePosition(Vector2 localPos) => Vector2.Divide(localPos, drawSize);
 
-            private Color4 colourAt(Vector2 localPos) => DrawColourInfo.Colour.HasSingleColour
-                ? (Color4)DrawColourInfo.Colour
-                : DrawColourInfo.Colour.Interpolate(relativePosition(localPos)).Linear;
+            private Color4 colourAt(Vector2 localPos) => DrawColourInfo.Colour.Interpolate(relativePosition(localPos)).Linear;
```
Comment (Tom94) [inline osu.Framework/Graphics/Lines/Path_DrawNode.cs]: I believe the branch previously existed for optimization. I don't think blindly removing it is a good idea, even though interpolation (obviously) gives the correct result, even if it happens between the same value.

File: osu.Framework/Graphics/Lines/Path_DrawNode.cs [hunk 4]
```diff
@@ -55,7 +55,9 @@ public override void ApplyState()
 
             private Vector2 relativePosition(Vector2 localPos) => Vector2.Divide(localPos, drawSize);
 
-            private Color4 colourAt(Vector2 localPos) => DrawColourInfo.Colour.Interpolate(relativePosition(localPos)).Linear;
+            private Color4 colourAt(Vector2 localPos) => DrawColourInfo.Colour.HasSingleColour
+                ? DrawColourInfo.Colour.AverageColour.Linear
```
Comment (KoziLord) [inline osu.Framework/Graphics/Lines/Path_DrawNode.cs]: `` ((SRGBColour)DrawColourInfo.Colour).Linear`` should also fix the issue while avoiding the check for the single colour from ``AverageColour``

Comment (jorolf) [reply osu.Framework/Graphics/Lines/Path_DrawNode.cs]: It doesn't avoid the single colour check:
https://github.com/ppy/osu-framework/blob/master/osu.Framework/Graphics/Colour/ColourInfo.cs#L220
https://github.com/ppy/osu-framework/blob/master/osu.Framework/Graphics/Colour/ColourInfo.cs#L71

**Pull request merged**